x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -565,6 +565,7 @@ private PhysicalOperation planHashJoin(HashJoinExec join, LocalExecutionPlannerC
 
     private PhysicalOperation planLookupJoin(LookupJoinExec join, LocalExecutionPlannerContext context) {
         PhysicalOperation source = plan(join.left(), context);
+        // TODO: The source builder includes incoming fields including the ones we're going to drop
```

Comment:
This comment was added very early on during discussions with you. We added it as a reminder. Could it be that it went stale immediately, since you did your fixes? This should be verified.

Reply:
It's always been stale, I just didn't understand the physical operators enough at the time :D

If you look [here](https://github.com/elastic/elasticsearch/blob/f2addbc69a2aa7fb512c1d5ca9a839f5fc7f5134/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexOperator.java#L123), you can see that the corresponding operator will only ever append blocks from the right.

I thought the layout was wrong because it would not take into account shadowing. But that's actually fine! Shadowing is a logical concept, the physical operators don't care. They only care about channels and which data is present in which. And for this, the layout constructed here appears to be very correct: it has the channels from the left hand side and additionally more channels slapped onto it from the right, containing the looked up fields.